timesketch/lib/llms/features/llm_summarize.py

```diff
@@ -0,0 +1,233 @@
+# limitations under the License.
+"""LLM Summarization feature."""
+import json
+import logging
+from typing import Any, Optional
+import pandas as pd
+import prometheus_client
+from flask import current_app
+from opensearchpy import OpenSearch
+from timesketch.lib import utils
+from timesketch.api.v1 import export
+from timesketch.models.sketch import Sketch
+from timesketch.lib.definitions import METRICS_NAMESPACE
+from timesketch.lib.llms.features.interface import LLMFeatureInterface
+
+logger = logging.getLogger("timesketch.llm.summarize_feature")
+
+# TODO(itsmvd): Remove 'feature' prefix after migration
+METRICS = {
+    "llm_summary_events_processed_total": prometheus_client.Counter(
+        "feature_llm_summary_events_processed_total",  # avoid duplicate registration
+        "Total number of events processed for LLM summarization",
+        ["sketch_id"],
+        namespace=METRICS_NAMESPACE,
+    ),
+    "llm_summary_unique_events_total": prometheus_client.Counter(
+        "feature_llm_summary_unique_events_total",  # avoid duplicate registration
+        "Total number of unique events sent to the LLM",
+        ["sketch_id"],
+        namespace=METRICS_NAMESPACE,
+    ),
+}
+
+
+class LLMSummarizeFeature(LLMFeatureInterface):
+    """LLM Summarization feature."""
+
+    NAME = "llm_summarize"
+    RESPONSE_SCHEMA = {
+        "type": "object",
+        "properties": {"summary": {"type": "string"}},
+        "required": ["summary"],
+    }
+
+    def _get_prompt_text(self, events_dict: list) -> str:
+        """Reads the prompt template from file and injects events.
+        Args:
+            events_dict: List of event dictionaries to inject into prompt.
+        Returns:
+            str: Complete prompt text with injected events.
+        Raises:
+            ValueError: If the prompt path is not configured.
+            FileNotFoundError: If the prompt file cannot be found.
+            IOError: If there's an error reading the prompt file.
+        """
+        prompt_file_path = current_app.config.get("PROMPT_LLM_SUMMARIZATION")
+        if not prompt_file_path:
+            logger.error("PROMPT_LLM_SUMMARIZATION config not set")
+            raise ValueError("LLM summarization prompt path not configured.")
+        try:
+            with open(prompt_file_path, "r", encoding="utf-8") as file_handle:
+                prompt_template = file_handle.read()
+        except FileNotFoundError as exc:
+            logger.error("Prompt file not found: %s", prompt_file_path)
+            raise FileNotFoundError(
+                f"LLM Prompt file not found: {prompt_file_path}"
+            ) from exc
+        except IOError as e:
+            logger.error("Error reading prompt file: %s", e)
+            raise IOError("Error reading LLM prompt file.") from e
+        prompt_text = prompt_template.replace("<EVENTS_JSON>", json.dumps(events_dict))
+        return prompt_text
+
+    def _run_timesketch_query(
```

Comment:
Running a Timesketch query seems like something more features would need to do in the future. Any plans to add this to the interface later on?
You could also offer something similar to the [event_stream](https://github.com/google/timesketch/blob/master/timesketch/lib/analyzers/interface.py#L1038) we use with the analyzer interface for searches exceeding the maximum data to be returned with one search.

Reply:
Yes 100% planning to add this to the `LLMResource` or to another type of helper library. Was also why I was experimenting with event_stream / most efficient way to retrieve events yesterday.